### ISSUE TITLE
Show follow states across servers

### ISSUE THREAD

**Issue by jaredzimmerman** [mastodon/mastodon#19955]
### Pitch

When i view a user's public profile on their server instance (not my local server) the [Follow] button does not reflect my follow state for them

### Motivation

follow state show persist across serves

**Comment by trwnh** [comment]
There is no way for the remote website to know who you are, much less obtain any state. You would have to add support for cross-domain authentication/authorization, and then pass some token along when clicking a link and loading the remote website.

**Comment by jaredzimmerman** [comment]
yes, exactly, self imposed privacy limitation make even basic functionality "not possible" currently.

**Comment by ineffyble** [comment]
There's no realistic solution for this. It's the equivalent of expecting your personal email inbox to know who you emailed from your work email address yesterday.

**Comment by jaredzimmerman** [comment]
There are other folks on here talking about federated login across servers. I can think of at least one option based on that. where the first time you visit a profile on another server, you have the option of a 1 click "login" that creates a shadow account on that server to enable bi-directional communication to your home server for basic info like follow status (and blocklist details) which would also help with 1-click follow on subsequent visits to that server's profiles
